### ISSUE TITLE
Add a simpler means of using Deed dice

### ISSUE THREAD

**Issue by algnc** [foundryvtt-dcc/dcc#215]
Some folks find the Deed dice system hard to follow. Maybe they forget to click it because it is separate from attacks. Maybe an option to prompt before an attack roll if you have selected Mighty Deeds on your sheet config?

**Comment by CyricPL** [comment]
If I could offer two cents here - if at all possible, I would like to see it account for a character being in an active encounter. Outside of an encounter, prompt the player to ask if a Deed roll is wanted every time they roll the action die. During an encounter, automatically roll Deed when first action die is rolled and apply the same result to all action die rolls until the encounter advances to the next turn.

**Comment by TylerKostuch** [comment]
For my warrior's attack macros, I've altered them to automatically roll the deed die for her.

Here's an example of what it looks like:
```
const _actor = game.dcc.getMacroActor(); 
if (_actor) { 
  _actor.rollAttackBonus();
  game.dcc.rollDCCWeaponMacro("m2", {"backstab":false});
}
```

Just drag one of the attacks down to the macro bar and alter it so that it rolls the attack bonus programmatically.

**Comment by CyricPL** [comment]
> For my warrior's attack macros, I've altered them to automatically roll the deed die for her.
> 
> Here's an example of what it looks like:
> 
> ```
> const _actor = game.dcc.getMacroActor(); 
> if (_actor) { 
>   _actor.rollAttackBonus();
>   game.dcc.rollDCCWeaponMacro("m2", {"backstab":false});
> }
> ```
> 
> Just drag one of the attacks down to the macro bar and alter it so that it rolls the attack bonus programmatically.

It actually just occurred to me to do the same this very week!

**Comment by radoslawg** [comment]
Pull Request #267 is my attempt at improving deed die mechanic. I understand why Deed Die in DCC Ruleset is implemented as is, and I understand developers do not want to "take stance" (quoted from Discord). What I did is, I gave options to groups that want to roll Deed Die with every attack. 

Reason behind this is that if player forgets to roll die with each round then value from previous round is added to results, it takes group out of action to recalculate manually rolls (granted, not complicated math but it takes a bit of time). Also, it gives option to groups that understands that core book rules mean to roll deed die with every attack.

This pull request also integrates deed die with DCC roll result cards. I tried to follow what I see is implemented already but javascript is not language I use every day.

**Comment by mooped** [comment]
For @TylerKostuch and @CyricPL 
> 
> 
> For my warrior's attack macros, I've altered them to automatically roll the deed die for her.
> 
> Here's an example of what it looks like:
> 
> ```
> const _actor = game.dcc.getMacroActor(); 
> if (_actor) { 
>   _actor.rollAttackBonus();
>   game.dcc.rollDCCWeaponMacro("m2", {"backstab":false});
> }
> ```
> 
> Just drag one of the attacks down to the macro bar and alter it so that it rolls the attack bonus programmatically.

This macro will have been broken for a few versions as actor.rollAttackBonus() did not wait for the last rolled attack bonus to update before returning. That is fixed on the dev-v0.30.0 branch now (and will be fixed in main and released packages once we release v0.30.0).

You will still need to add an 'await' before _actor.rollAttackBonus(), but the rolls will now happen in the correct order and the attack roll will use the new attack bonus instead of the previous.

I'm also looking at integrating a version of the suggestions @radoslawg contributed in his pull request so that might not be necessary soon anyway.

**Comment by mooped** [comment]
I've done some rework on this in the following commit if you're interested:

https://github.com/foundryvtt-dcc/dcc/commit/571333697bd08acdbf3215b339440c2f1d49217e

I've switched it to use a dropdown to select a mode, which should futureproof things and reduce the number of combinations. I've also reworked the emotes to reduce the number of combinations there.

I need to do some more testing (and fix the tests), but I'm out of time for now.

![image](https://user-images.githubusercontent.com/371357/129273013-4e310f47-8a8f-4246-b4d7-a05c476a8f13.png)

The behaviour is essentially identical to @radoslawg 's change.
